src/auth/feedback-secret.guard.ts

```diff
@@ -0,0 +1,24 @@
+import {
+  Injectable,
+  CanActivate,
+  ExecutionContext,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { Env } from 'src/env';
+
+@Injectable()
+export class FeedbackSecretGuard implements CanActivate {
+  constructor() {}
+
+  canActivate(context: ExecutionContext): boolean {
+    const request = context.switchToHttp().getRequest();
+    const secret = Env.SECRET.FEEDBACK;
+    const requestSecret = request.headers['x-feedback-secret'];
+
+    if (requestSecret && requestSecret === secret) {
+      return true;
+    } else {
+      throw new UnauthorizedException('Invalid feedback secret');
+    }
```

Comment:
PullProofAI

It's recommended to provide a more detailed error message for better debugging and understanding.
```diff
-       throw new UnauthorizedException('Invalid feedback secret');
+       throw new UnauthorizedException('Access denied: Invalid feedback secret provided');
```

### Summary
The code changes look good overall with minor improvements suggested for readability and error handling.


<!-- This is an auto-generated comment by PullProofAI -->